homeassistant/components/input_number.py

```diff
@@ -19,18 +19,25 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-DOMAIN = 'input_slider'
+DOMAIN = 'input_number'
 ENTITY_ID_FORMAT = DOMAIN + '.{}'
 
 CONF_INITIAL = 'initial'
 CONF_MIN = 'min'
 CONF_MAX = 'max'
+CONF_MODE = 'mode'
 CONF_STEP = 'step'
 
+MODE_BOTH = 'both'
+MODE_SLIDER = 'slider'
+MODE_BOX = 'box'
+
 ATTR_VALUE = 'value'
 ATTR_MIN = 'min'
 ATTR_MAX = 'max'
 ATTR_STEP = 'step'
+ATTR_HIDDEN_SLIDER = 'hiddenslider'
```

Comment:
We should not store this as attributes. Instead, this should be computed on the frontend based on the mode.